party_time/party_time.lua

```diff
@@ -0,0 +1,345 @@
+        elseif alias.off[str] then
+            return false
+        end
+
+        error('Expected a boolean value.')
+    end
+})
+
+-- Addon Command Handlers
+-- Seven main commands additoinal sub_commands within
+local pt_command = command.new("pt")
+
+function join(name)
+    command.input("/prcmd add " .. name)
+end
+
+pt_command:register("j", join, '<name:string>')
+pt_command:register("join", join, '<name:string>')
+
+
+function invite(...)
+    for _, name in pairs({...}) do
+        command.input("/pcmd add " .. name)
+        name = name:gsub("^%l", string.upper)
+        if unhandled_requests[name] then
+            unhandled_requests[name] = nil
+        end
+    end
+end
+
+pt_command:register("i", invite, '{names}')
+pt_command:register("invite", invite, '{names}')
+
+
+function blacklist(section, sub_cmd, ...)
+    local names = {...}
+    if alias.add[sub_cmd] then
+        for _, name in pairs(names) do
+            pt_settings.blacklist[section][name] = true
+        end
+    elseif alias.remove[sub_cmd] then
+        for _, name in pairs(names) do
+            pt_settings.blacklist[section][name] = nil
+        end
+    end
+    settings.save(pt_settings)
+end
+
+pt_command:register("b", blacklist, '<sub_cmd:one_of(add,a,+,remove,rm,r,-)> {names}')
+pt_command:register("blacklist", blacklist, '<sub_cmd:one_of(add,a,+,remove,rm,r,-)> {names}')
+
+
+function whitelist(sub_cmd, ...)
+    local names = {...}
+    if alias.add[sub_cmd] then
+        for _, name in pairs(names) do
+            pt_settings.whitelist[name] = true
+        end
+    elseif alias.remove[sub_cmd] then
+        for _, name in pairs(names) do
+            pt_settings.whitelist[name] = nil
+        end
+    end
+    settings.save(pt_settings)
+end
+
+pt_command:register("w", whitelist, '<sub_cmd:one_of(add,a,+,remove,rm,r,-)> {names}')
+pt_command:register("whitelist", whitelist, '<sub_cmd:one_of(add,a,+,remove,rm,r,-)> {names}')
+
+
+function ui_enable(bool)
+    pt_settings.ui.enabled = bool
+    settings.save(pt_settings)
+end
+
+pt_command:register("ui_enable", ui_enable, '<enabled:boolean>')
+
+
+function auto_accept_enable(bool)
+    pt_settings.auto.accept_invites = bool
+    settings.save(pt_settings)
+end
+
+pt_command:register("auto_accept", auto_accept_enable, '<enabled:boolean>')
+
+
+function auto_decline_enable(bool)
+    pt_settings.auto.decline_invites = bool
+    settings.save(pt_settings)
+end
+
+pt_command:register("auto_decline", auto_decline_enable, '<enabled:boolean>')
+
+-- Packet Event Handlers
+-- Recieve Invite & Recieve Request
+packets.incoming[0x0DC]:register(function(p)
+    if pt_settings.auto.accept_invites and pt_settings.whitelist[p.player_name] then
+        coroutine.schedule(function()
+            local clock = os.clock()
+            repeat
+                if (os.clock() - clock) > 90 then
+                    return
+                end
+                coroutine.sleep_frame()
+            until(#treasure == 0)
```

Comment:
I'm not sure how the treasure lib is set up, but will this work if an item is in a higher order slot?

Reply:
treasure has a __len meta that use pairs to get a full count, in case that happens